src/Raven.Server/Integrations/PostgreSQL/PgServer.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Server.Config;
+using Raven.Server.Config.Categories;
+using Raven.Server.Exceptions;
+using Sparrow.Logging;
+
+namespace Raven.Server.Integrations.PostgreSQL
+{
+    public class PgServer : IDisposable
+    {
+        private readonly Logger _logger = LoggingSource.Instance.GetLogger<PgServer>("Postgres Server");
+
+        private readonly RavenServer _server;
+        private readonly ConcurrentDictionary<TcpClient, Task> _connections = new();
+        private readonly CancellationTokenSource _cts = new();
+        private readonly SemaphoreSlim _locker = new SemaphoreSlim(1, 1);
+        private readonly ConcurrentQueue<TcpListener> _listeners = new ConcurrentQueue<TcpListener>();
+        private readonly int _processId;
+        private readonly int _port;
+
+        private int _sessionIdentifier;
+        private bool _disposed;
+        
+        public PgServer(RavenServer server)
+        {
+            _server = server;
+            _processId = Process.GetCurrentProcess().Id;
+            _port = _server.Configuration.Integrations.PostgreSql.Port;
+
+            _server.ServerStore.LicenseManager.LicenseChanged += HandleServerActivation;
+        }
+
+        public bool Active { get; private set; }
+
+        public void Execute()
+        {
+            HandleServerActivation();
+        }
+
+        private void HandleServerActivation()
+        {
+            if (_server.Configuration.Integrations.PostgreSql.Enabled == false)
+                return;
+
+            _locker.Wait();
+
+            try
+            {
+                bool activate = false;
+
+                if (_server.ServerStore.LicenseManager.CanUsePowerBi(withNotification: false, out _))
+                {
+                    activate = true;
+                }
+                else if (_server.ServerStore.LicenseManager.CanUsePostgreSqlIntegration(withNotification: true))
+                {
+                    if (_server.ServerStore.Configuration.Core.FeaturesAvailability == FeaturesAvailability.Experimental)
+                        activate = true;
+                    else
+                    {
+                        if (_logger.IsOperationsEnabled)
+                            _logger.Info($"You have enabled the PostgreSQL integration via '{RavenConfiguration.GetKey(x => x.Integrations.PostgreSql.Enabled)}' configuration but " +
```

Comment:
Operations here, not info

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/12989/commits/316d280ae7e484305a19d3051db1ecc64cd68a48